app/controllers/products_controller.rb

```diff
@@ -50,4 +63,18 @@ def set_product!
   def product_params
     params.require(:product).permit(:name, :image, :code, :price, :description, :remove_image)
   end
+
+  def page_limit
```

Comment:
rename to max_page(products_count) and move to concerns https://www.bigbinary.com/learn-rubyonrails-book/keep-application-controller-clean.